turborepo-tests/integration/tests/config.t

```diff
@@ -0,0 +1,399 @@
+      "check_for_update": false,
+      "test_run": true,
+      "run_args": null,
+      "command": {
+        "Run": {
+          "cache_dir": null,
+          "cache_workers": 10,
+          "concurrency": null,
+          "continue_execution": false,
+          "dry_run": null,
+          "single_package": false,
+          "filter": [],
+          "force": false,
+          "global_deps": [],
+          "graph": null,
+          "env_mode": "Infer",
+          "ignore": [],
+          "include_dependencies": false,
+          "no_cache": false,
+          "no_daemon": false,
+          "no_deps": false,
+          "output_logs": null,
+          "only": false,
+          "parallel": false,
+          "pkg_inference_root": "",
+          "profile": null,
+          "remote_only": false,
+          "scope": [],
+          "since": null,
+          "summarize": null,
+          "log_prefix": null,
+          "tasks": [
+            "build"
+          ],
+          "pass_through_args": [],
+          "experimental_space_id": null
+        }
+      }
+    }
+  }
+
+Run test run with team overloaded
+  $ ${TURBO} run build --__test-run --team vercel
+  Invoking go binary with {
+    "remote_config": {
+      "token": null,
+      "team_id": null,
+      "team_slug": "vercel",
+      "api_url": "https://vercel.com/api"
+    },
+    "cli_args": {
+      "version": false,
+      "no_update_notifier": false,
+      "api": null,
+      "color": false,
+      "cpu_profile": null,
+      "cwd": ".*", (re)
+      "heap": null,
+      "login": null,
+      "no_color": false,
+      "preflight": false,
+      "remote_cache_timeout": null,
+      "team": "vercel",
+      "token": null,
+      "trace": null,
+      "verbosity": 0,
+      "check_for_update": false,
+      "test_run": true,
+      "run_args": null,
+      "command": {
+        "Run": {
+          "cache_dir": null,
+          "cache_workers": 10,
+          "concurrency": null,
+          "continue_execution": false,
+          "dry_run": null,
+          "single_package": false,
+          "filter": [],
+          "force": false,
+          "global_deps": [],
+          "graph": null,
+          "env_mode": "Infer",
+          "ignore": [],
+          "include_dependencies": false,
+          "no_cache": false,
+          "no_daemon": false,
+          "no_deps": false,
+          "output_logs": null,
+          "only": false,
+          "parallel": false,
+          "pkg_inference_root": "",
+          "profile": null,
+          "remote_only": false,
+          "scope": [],
+          "since": null,
+          "summarize": null,
+          "log_prefix": null,
+          "tasks": [
+            "build"
+          ],
+          "pass_through_args": [],
+          "experimental_space_id": null
+        }
+      }
+    }
+  }
+
+Run test run with team overloaded from both env and flag (flag should take precedence)
+  $ TURBO_TEAM=vercel ${TURBO} run build --__test-run --team turbo
```

Comment:
One more test for both `TURBO_TOKEN` and `--token` and `VERCEL_ARTIFACTS_TOKEN` would be awesome